test/extended/prometheus/prometheus_builds.go

```diff
@@ -209,10 +215,10 @@ func labelsWeWant(sample *model.Sample, labels map[string]string) bool {
 }
 
 func valueWeWant(sample *model.Sample, tc metricTest) bool {
-	//NOTE - we could use SampleValue has an Equals func, but since SampleValue has no GreaterThan,
+	//NOTE - we could use SampleValue has an Equals func, but since SampleValue has no GreaterThanEqual,
 	// we have to go down the float64 compare anyway
-	if tc.greaterThan {
-		return float64(sample.Value) > tc.value
+	if tc.greaterThanEqual {
+		return float64(sample.Value) >= tc.value
 	}
-	return float64(sample.Value) == tc.value
+	return float64(sample.Value) < tc.value
```

Comment:
seems slightly odd that you're changing `==` to `<` here, not least given AFAICS there is no metricTest which doesn't have greaterThanEqual set, so this code path isn't actually ever used?

Reply:
if you didn't explicitly request greaterThanEqual, then you're implicitly requesting less than... (at least under the new semantics introduced by this PR).

Reply:
correct that the code path is not used currently; I wanted the code in place to facilitate usage of greaterThanEqual not being set in the future

and yes, given the precise values are not guarantee-able when this test is run in parallel / in conjunction with other tests, I wanted to have either "greater than or equal" or "less than or equal".